File: Selenium/Activities/Activity15.java
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class Activity15 {
    public static void main(String[] args) {
        WebDriver driver = new FirefoxDriver();
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(20));

        driver.get("https://v1.training-support.net/selenium/dynamic-attributes");
        System.out.println(driver.getTitle());

        WebElement element1 = driver.findElement(By.xpath("//input[starts-with(@class, 'username-')]"));
        element1.sendKeys("admin");

        WebElement element2 = driver.findElement(By.xpath("//input[starts-with(@class, 'password-')]"));
        element2.sendKeys("password");

        driver.findElement(By.xpath("//*[@id=\"dynamic-attributes-form\"]/div/button")).click();

        wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//*[@id=\"action-confirmation\"]")));
        System.out.println(driver.findElement(By.xpath("//*[@id=\"action-confirmation\"]")).isDisplayed());

        System.out.println(driver.findElement(By.xpath("//*[@id=\"action-confirmation\"]")).getText());

        driver.quit();

    }
}

File: Selenium/Activities/Activity15.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.wait import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

driver = webdriver.Firefox()
wait = WebDriverWait(driver, 10)

driver.get("https://v1.training-support.net/selenium/dynamic-attributes")
print(driver.title)

driver.find_element(By.XPATH, "//input[starts-with(@class, 'username-')]").send_keys("admin")
driver.find_element(By.XPATH, "//input[starts-with(@class, 'password-')]").send_keys("password")
driver.find_element(By.XPATH, '//button[@type = "submit"]').click()

wait.until(EC.visibility_of_element_located((By.XPATH, '//*[@id="action-confirmation"]')))
print(driver.find_element(By.XPATH, '//*[@id="action-confirmation"]').text)

driver.close()


